common/src/main/resources/common.properties

```diff
@@ -0,0 +1,12 @@
+spring.rabbitmq.host=localhost
```

Comment:
I'm nor sure if it would work, but you can create an `application-common.yml` file to use it as a Spring profile you can include in your other modules. The `@ConfigurationProperties` class that you have should work in the same way without needing to use the `@PropertySource` annotation.

Reply:
In the `common` library I am using a different file name so that it can be imported into other modules by using `@PropertySource("classpath:common.properties")`. This way, I have common properties defined in this module and they can be overriden by the corresponding `application.yml`.

The name `common.properties` also, from my perspective, makes it clearer that this is a shared properties file and not just used for the module it is stored in.